CMakeLists.txt

```diff
@@ -212,7 +217,7 @@ ELSE()
   SET(security_default ON)
 ENDIF()
 OPTION(SECURITY_HARDENED "Use security-enhancing compiler features (stack protector, relro, etc)" ${security_default})
-IF(SECURITY_HARDENED)
+IF(SECURITY_HARDENED AND NOT WITH_TSAN)
```

Comment:
Yeah, that's almost what I meant. It disables SECURITY_HARDENED implicitly, which is alright. Unless SECURITY_HARDENED was specified by user. I suggest the following updates (sorry that I didn't suggest them in the first review)...

Under ```IF(WITH_TSAN)``` before setting flags:
```
IF(SECURITY_HARDENED)
    MESSAGE(FATAL_ERROR "WITH_TSAN and SECURITY_HARDENED are mutually exclusive")
ENDIF()
```

Move ```AND NOT WITH_TSAN``` up:
```
IF(CMAKE_C_COMPILER_VERSION VERSION_LESS "4.6" OR WITH_TSAN)
  SET(security_default OFF)
ELSE()
  SET(security_default ON)
ENDIF()
```

Then this patch should be alright.